### DISCUSSION TITLE
How to write an "always transition" followed by a conditional transition?

### DISCUSSION THREAD

**Discussion by smac89** [statelyai/xstate#3031] [Q&A] [answered]
Let's say I have an event `pathChanged`.

Now I want to update the context using an action like:

```js
{
   ...
   on: {
     pathChanged: [
        {
            actions: assign({isProtected: (_ctx, {requiresAuth}) => requiresAuth ?? false})
        },
        {
            target: 'authenticating',
            cond: (...args) => !isAuthenticated(...args) && isRouteProtected(...args)
        }
     ]
  }
}
```

What I would like to achieve is for the first transition to **always** be taken, then the subsequent ones are tested against the condition.

However, right now it just takes the first transition, and the second is never taken.

Is this possible? Is there another way to do this?

Thanks!

**Comment by davidkpiano** [comment, marked as answer]
Right now, you have to duplicate the action in both transitions. Because of how it's structured, there's no DRY way to do this currently, but we're open to ideas.

**Reply by smac89** [reply]
I had a feeling this was going to be the only way, and that's what I ended up doing.

In order to avoid repeating myself too much, I extracted the action to be a function on it's own, and simply added that function in both places.

Thanks for the feedback!

**Reply by davidkpiano** [reply]
That's a good, idiomatic approach 💯
